From: Alec Leamas <leamas@nowhere.net>
Date: Sat, 28 Jul 2018 13:04:46 +0200
Subject: Unix: Add a simple manpage.

Origin: upstream https://github.com/OpenCPN/OpenCPN/commit/e1f621a0d9d5
Applied-Upstream: 5.0.0

Adding a manpage is mostly to have a pointer to the documentation on the
web in a standard location. However, it's also a good place for
Linux/unix-specific power user info.

Also, Debian requires a manpage.
---
 CMakeLists.txt |  2 ++
 opencpn.1      | 78 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 80 insertions(+)
 create mode 100644 opencpn.1

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 46effb1..ca76ccd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2090,6 +2090,8 @@ IF(UNIX)
   INSTALL(FILES data/opencpn.png DESTINATION ${PREFIX_DATA}/icons/hicolor/48x48/apps PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
   INSTALL(FILES src/bitmaps/other_svg_src/opencpn.svg DESTINATION ${PREFIX_DATA}/icons/hicolor/scalable/apps PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
   INSTALL(FILES data/opencpn.desktop DESTINATION ${PREFIX_DATA}/applications PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ )
+  INSTALL(FILES data/opencpn.appdata.xml DESTINATION ${PREFIX_DATA}/appdata PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ )
+  INSTALL(FILES opencpn.1 DESTINATION ${PREFIX_DATA}/man/man1 PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ )
 ENDIF(UNIX)
 
 ENDIF(NOT APPLE)
diff --git a/opencpn.1 b/opencpn.1
new file mode 100644
index 0000000..550501a
--- /dev/null
+++ b/opencpn.1
@@ -0,0 +1,78 @@
+.\"
+.\" (C) Copyright 2018 Alec Leamas <leamas.alec@gmail.com>,
+.\"
+.TH Opencpn 1 "July 2018"
+
+.SH NAME
+opencpn \- Chartplotter and GPS Navigation Software
+.SH SYNOPSIS
+.B opencpn
+.RI [ options ]
+.SH DESCRIPTION
+OpenCPN is a chartplotter and navigation GUI program supporting navigation
+under way as well as planning. It supports a wide range of chart formats
+using a plugin system. Other plugins provides support for e. g. radar, AIS
+and weather maps. Program is localized in 20+ languages and ported to
+several OS including MacOS, Windows and Linux.
+
+.SH OPTIONS
+.TP
+.B \-h, \-\-help
+Show help info.
+.TP
+.B \-p
+Run in portable mode.
+.TP
+.B  \-fullscreen
+Switch to full screen mode on start.
+.TP
+.B  \-no_opengl
+Disable OpenGL video acceleration. This setting will be remembered.
+.TP
+.B  \-rebuild_gl_raster_cache
+Rebuild OpenGL raster cache on start.
+.TP
+.B  \-parse_all_enc
+Convert all S-57 charts to OpenCPN's internal format on start.
+.TP
+.B  \-unit_test_1:<num>
+Display a slideshow of <num> charts and then exit. Zero or negative <num>
+specifies no limit.
+
+.SH PLUGIN LOADING
+
+Plugin loading depends on that the opencpn main program can locate the
+plugin object file and that the plugin can locate it's own data.
+.P
+On Linux, the main program searches for plugin .so-files in a library
+named \fIopencpn/plugins\fR.
+By default, this path is searched in ~/.local/lib:/usr/local/lib:/usr/lib.
+The default search path can be overridden by the OPENCPN_PLUGIN_DIRS
+environment variable.
+.P
+On other platforms, all plugin object files lives in a single directory
+defined at installation.
+.P
+A plugin's data directory is a path ending with
+\fIopencpn/plugins/<plugin_name>\fR.
+On Linux, this is by default searched in
+~/local/share:/usr/local/share:/usr/share.
+This search path is overridden by XDG_DATA_DIRS as defined by the freedesktop
+spec.
+Only plugins adapted for this (i. e., using GetPluginDataDir()) handles this
+properly.
+.P
+On other platforms, the plugin data directories always lives in a common
+parent defined at installation time.
+
+.SH NOTES
+
+The support for OPENCPN_PLUGIN_DIRS and XDG_DATA_DIRS was introduced in
+version TBD. Prior to this, all plugins was loaded from a single directory
+and all plugin data directories had a common parent.
+
+.SH SEE ALSO
+.br
+http://www.opencpn.org
+.br
+http://www.freedesktop.org
